apiserver/pkg/server/cluster_server.go

```diff
@@ -141,6 +141,10 @@ func ValidateCreateClusterRequest(request *api.CreateClusterRequest) error {
 		return util.NewInvalidInputError("Namespace is empty. Please specify a valid value.")
 	}
 
+	if request.Namespace != request.Cluster.Namespace {
```

Comment:
It looks good to me. 

Let's also add the check in `compute_template_server.go`

Reply:
Yup it's added.